### PR TITLE
RS-626 A-Frame Implementation

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

**NOTE: 
1. Still needs to be tested by being deployed behind https security DONE
2. Tested with Oculus Quest DONE
4. and Iphone devices DONE
5. Test items with both VR and AR assets DONE
6. Test with Android device DONE
7. Ensure storybook is working and documented DONE
8. Error on Iphone when exitting screen.  Image is not fullsized. DONE
9. The exit button disappears when going fullscreen mode.  DONE
10.  Change the loading screen on IOS. DONE

Linting:

- [x] No linting errors

Component Updates:

- [x] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:

- [x] E2E tests (npm test run with `e2e`)
- [x] Manual tests

Documentation:

- [x] Requires documentation updates

### CODE REVIEW

File: src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+          <Scene className={`vr-container ${overlayScene ? 'overlay' : 'underlay'}`} embedded vr-mode-ui="enterVRButton: #myEnterVRButton;">
+            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+            <a id="myEnterVRButton" href="#">
+              {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
+              <div role="button" tabIndex={0} className="vr-fullscreen" onClick={() => { this.setOverlay(); }} />
+            </a>
+
+            <a-assets>
+              <img alt="" id="luxuryCar" src={backgroundUri} />
```
Comment (shaunmaharaj) [inline src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: Looks like a-frame is throwing a warning about the element requested without `crossorigin`. 
Let's look into if this helps: ``

**Comment by shaunmaharaj** [comment]
Completed testing of the following:
- [X] Still needs to be tested by being deployed behind https security 
- [X] Tested with Oculus Quest and Iphone devices

File: src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx (hunk 1 above)
Comment (shaunmaharaj) [inline src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]:
```suggestion
              <img alt="" id="luxuryCar" src={backgroundUri} crossOrigin="anonymous" />
```

Comment (shaunmaharaj) [inline src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: Let's change the id to something more generic

Comment (aChanEP) [reply src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: WIP

Comment (aChanEP) [reply src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: WIP

Comment (aChanEP) [reply src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: resolved and will work for both safari, chrome and firefox

File: src/components/src/VRProductDisplayItem/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#### Usage
+
+```js
+import { VRProductDisplayItem } from '@elasticpath/store-components';
+```
+
+#### Example
+
+```js
+<VRProductDisplayItem handleCloseVR={() => { console.log('handlecloseVR) }} backgroundUri="https://s3.amazonaws.com/refexp/vr/1234.jpg" />
```
Comment (shaunmaharaj) [inline src/components/src/VRProductDisplayItem/README.md]: The url in this example doesn't actually resolve to an image. Just FYI

File: src/ep.config.json [hunk 3]
```diff
@@ -1,9 +1,7 @@
 {
   "cortexApi": {
-    "path": "/cortex",
-    "scope": "vestri",
-    "pathForProxy": "http://localhost:9080",
-    "reqTimeout": "30000"
+    "path": "https://reference.epdemos.com/cortex",
```
Comment (shaunmaharaj) [inline src/ep.config.json]: These configs shouldn't be set OOTB

File: src/ep.config.json [hunk 4]
```diff
@@ -27,13 +25,9 @@
     }
   ],
   "defaultCurrencyValue": "CAD",
-  "skuImagesUrl": "https://elasticpath-demo-images.s3-us-west-2.amazonaws.com/VESTRI_VIRTUAL_V1/%sku%.png",
-  "siteImagesUrl": "https://elasticpath-demo-images.s3-us-west-2.amazonaws.com/VESTRI_VIRTUAL_V1/%fileName%",
+  "skuImagesUrl": "https://elasticpath-demo-images.s3-us-west-2.amazonaws.com/VESTRI_VIRTUAL_V2/%sku%.png",
```
Comment (shaunmaharaj) [inline src/ep.config.json]: We still need to continue using V1 for now due to the catalog

File: src/ep.config.json [hunk 5]
```diff
@@ -27,13 +25,9 @@
...
+  "siteImagesUrl": "https://elasticpath-demo-images.s3-us-west-2.amazonaws.com/VESTRI_VIRTUAL_V2/%fileName%",
   "enableOfflineMode": false,
-  "imageFileTypes": {
```
Comment (shaunmaharaj) [inline src/ep.config.json]: accidental removal?

File: src/ep.config.json [hunk 6]
```diff
@@ -42,7 +36,8 @@
       "reqTimeout": "30000"
     },
     "openId": {
-      "enable": true
+      "enable": true,
+      "callbackUrl": "http://localhost:8080"
     },
     "keycloak": {
       "callbackUrl": "http://localhost:8080",
```
Comment (shaunmaharaj) [inline src/ep.config.json]: This shouldn't be reverted

File: storybook/custom_typings/aframe.d.ts [hunk 7]
```diff
@@ -0,0 +1,318 @@
+// Type definitions for AFRAME v0.3.1
+// Project: https://aframe.io/
+// Definitions by: Paul Shannon <https://github.com/devpaul/aframe-typings>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace THREE {
```
Comment (shaunmaharaj) [inline storybook/custom_typings/aframe.d.ts]: This needs to be created at the storybook level too?

File: src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+          <Scene className="vr-container" embedded vr-mode-ui="enterVRButton: #myEnterVRButton;" loading-screen="backgroundColor: #000000; dotsColor: white">
+            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+            <a id="myEnterVRButton" href="#">
+              {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
+              <div role="button" tabIndex={0} className="vr-fullscreen" />
+            </a>
+
+            <a-assets>
+              {
+                VRProductDisplayItem.isSafari() ? (
```
Comment (shaunmaharaj) [inline src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx]: 😬

File: src/components/src/VRProductDisplayItem/README.md (hunk 2 above)
Comment (aChanEP) [reply src/components/src/VRProductDisplayItem/README.md]: Yeah ill change that.  For some reason I thought we wouldn't want a live url? But i dont see any drawback

File: src/ep.config.json (hunk 3 above)
Comment (aChanEP) [reply src/ep.config.json]: will revert `ep.config.json`

**Pull request merged**